presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -50,13 +53,12 @@
         private final PlanNodeId planNodeId;
         private final List<Type> sourceTypes;
         private final List<Integer> outputChannels;
-        private final List<WindowFunctionDefinition> windowFunctionDefinitions;
+        private final Map<WindowFunctionDefinition, FrameInfo> windowFunctionDefinitions;
```

Comment:
Why don't we just make the FrameInfo part of the WindowFunctionDefinition? Or maybe put it into a container class called FramedWindowFunctionDefinition? That seems better than using a map here.